sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -535,7 +535,8 @@ case class Range(
 case class Aggregate(
     groupingExpressions: Seq[Expression],
     aggregateExpressions: Seq[NamedExpression],
-    child: LogicalPlan)
+    child: LogicalPlan,
+    stateful: Boolean = false)
   extends UnaryNode {
```

Comment:
`stateful` indicates if the aggregate is base on streaming or batch, resolved by `ResolveStatefulAggregate` rule